### PR TITLE
[bitnami/postgresql] fix replication with nodePort service type  #2063 

### LINKED ISSUE #2063: [postgresql] NodePort + Relication fails
**Which chart**:
postgresql 8.6.4

**Description**
Running install with the following values fails:
```yaml
service:
  type: NodePort
  nodePort: 30042
replication:
  enabled: true
  slaveReplicas: 2
```
due to a collision in the nodePort for the master and slave services.

I'd propose a master.nodePort and slave.nodePort _but_ it occurs to me one might want master and slaves running different service types. We've also got that use case as master gets fed outside the cluster and the slaves serve internal workloads so given that I'd propose service entries under master and slave that override the top level service entry if present.

If that sounds reasonable I'll get to work on a PR.

Example proposed:
```yaml
service:
  type: NodePort
  nodePort: 30042
replication:
  enabled: true
  slaveReplicas: 2
slave:
  service:
    nodePort: 30043
```
alternatively expressible as:
```yaml
service:
  type: NodePort
replication:
  enabled: true
  slaveReplicas: 2
master:
  service:
    nodePort: 30042
slave:
  service:
    nodePort: 30043
```
More complex example:
```yaml
replication:
  enabled: true
  slaveReplicas: 2
master:
  service:
    type: NodePort
    nodePort: 30042
slave:
  service:
    type: ClusterIP
```

### PR DESCRIPTION
**Pull request by Git-Lucky**
**Description of the change**

When replication is enabled, if the service type is nodePort, there is a collision with the master and slave nodePorts. 

**Benefits**

This adds a master and slave service, along with type, clusterIP, and nodePort to be set if the master and slave should differ allowing for differing values which fixes nodePort collision as well as offering more flexibility in general

**Applicable issues**

Fixes #2063 for postgresql

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Comment by Git-Lucky** [comment]
Sorry about the new pull request...got a little out of sorts with the last one.

File: bitnami/postgresql/README.md [hunk 1]
```diff
@@ -275,6 +281,10 @@ This chart includes a `values-production.yaml` file where you can find some para
 
 To horizontally scale this chart, you can use the `--replicas` flag to modify the number of nodes in your PostgreSQL deployment. Also you can use the `values-production.yaml` file or modify the parameters shown above.
 
+### Customizing Master and Slave services in a replicated configuration
+
+At the top level, there is a service object which defines the services for both master and slave. For deeper customization, there are service objects for both the master and slave types individually. This allows you to override the values in the top level service object so that the master and slave can be of different service types and with different clusterIPs / nodePorts. Also in the case you want the master and slave to be of type nodePort, you will need to set the nodePorts to different values to prevent a collision. The values that are deeper in the master.service or slave.service objects will take precedence over the top level service object.
+
```
Comment (juan131) [inline bitnami/postgresql/README.md:287]: Fantastic! Thanks so much

**Comment by juan131** [comment]
In order to update the image tags to point to latest releases, our CI/CD needs to modify you PR (`bitnami-bot` will update `values.yaml` and `values-production.yaml` for you). To do so, it's necessary you allow "edits from maintainers". Find more information in the link below: 

- https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

File: bitnami/postgresql/README.md (hunk 1 above)
Comment (Git-Lucky) [reply bitnami/postgresql/README.md:287]: Glad to help :)

**Comment by Git-Lucky** [comment]
Hi @juan131, Im attempting to allow edits and I saw the linked article but Im not having any luck finding it on this page.

**Comment by juan131** [comment]
As an alternative you can add `bitnami-bot` account as collaborator to your repos' fork. Ping me when that's done and I'll trigger the process

**Comment by odensc** [comment]
This PR seems to have broken the chart for me (when replication is enabled):
```
executing "charts/postgresql/templates/svc-read.yaml" at <$service>: undefined variable: $service
```

**Comment by juan131** [comment]
@odensc we created another PR to fix that issue. See https://github.com/bitnami/charts/pull/2241

**Pull request merged**